src/main/java/javax/jmdns/impl/ServiceInfoImpl.java

```diff
@@ -1165,6 +1173,7 @@ public String getNiceTextString() {
     @Override
     public ServiceInfoImpl clone() {
         ServiceInfoImpl serviceInfo = new ServiceInfoImpl(this.getQualifiedNameMap(), _port, _weight, _priority, _persistent, _text);
+        serviceInfo.setServer(_server);
```

Comment:
All other fields (besides the IP addresses) are set through the constructor and the constructor also already sets the `_server` field - so it looks a bit weird here to set this through a different mechanism. Wdyt?

Reply:
I had considered adding it to the constructor but there is a little bit of weirdness here that drove me to do it this way. While true that the constructor initializes `_server` using the value of `text`, the value can then be updated when processsing the SRV record. If that has happened prior to calling this `clone()` we end up with a clone that isn't quite the same thing. I also can't just put the value of `_server` as the value in the `text` parameter since that value is encoded in the constructor and we could end up with a double encoded string.